### DISCUSSION TITLE
Discussion / resources on prose diffs

### DISCUSSION THREAD

**Discussion by rprimet** [CatalaLang/catleg#9] [General]
I think Catala programs are limited to 80 chars line width, so reference texts as found in e.g. Legifrance are usually manually reformatted.

This discussion is about how we can still get useful diffs in that context ; 

One possibility is to reformat both texts to use one line per sentence.

This HN thread discusses that solution: https://news.ycombinator.com/item?id=13985427

Another idea might be to have an automatic reformatter that takes Legifrance texts and reformats those to 80 characters, and adopts some markdown conventions etc., a bit like the black formatter for python. This would be more complex obviously

An important aspect is to keep track of the line:character info whenever possible, so that the diffs will point the programmer in the right direction :-)

Thoughts / ideas welcome!

**Comment by denismerigoux** [comment]
Isn't https://github.com/davidar/pandiff exactly what we want?

**Comment by rprimet** [comment]
I think we can close that discussion for now, as we have a working diff, although it still needs improvements (especially for tables).
